### ISSUE TITLE
Incoming call not working in Ionic application

### ISSUE THREAD

**Issue by flitwebsgit** [jefflinwood/cordova-plugin-twiliovoicesdk#62]
Hi, @jefflinwood currently I'm facing an issue that is incoming call is not working I mean the outgoing call is working fine but if I'm calling from one device to another of different identity access token than incoming to that device is not working, no call notification or the dialog is appearing.

**Comment by jefflinwood** [comment]
Hi,

It's probably a problem with the FCM integration - Firebase Cloud Messaging - do you have the push plugin receiving the FCM messages?

**Comment by flitwebsgit** [comment]
@jefflinwood yes i have push plugin and when i try to publish notification from firebase  all notification is receiving to the devices

**Comment by flitwebsgit** [comment]
![Screenshot 2019-08-14 at 10 50 12 AM](https://user-images.githubusercontent.com/40888625/62996207-6e2cd000-be81-11e9-8ad1-7b6e68e1097c.png)
above is my project.properties file

![Screenshot 2019-08-14 at 10 50 45 AM](https://user-images.githubusercontent.com/40888625/62996233-8ac90800-be81-11e9-90b8-bb5b010e33b3.png)
and thisone is my config.xml

**Comment by jayzam** [comment]
I was having the same issue with incoming calls not working on ios.  (It works on Android)  I found this from Twilio and it helped me to resolve the problem.

https://github.com/twilio/twilio-voice-ios/blob/Releases/iOS-13-Migration-Guide.md

Based on this I made a change to TwilioVoicePlugin.m in this function.

#pragma mark PKPushRegistryDelegate methods
- (void)pushRegistry:(PKPushRegistry *)registry didUpdatePushCredentials:(PKPushCredentials *)credentials forType:(PKPushType)type {
    if ([type isEqualToString:PKPushTypeVoIP]) {

        const unsigned *tokenBytes = [credentials.token bytes];
        self.pushDeviceToken = [NSString stringWithFormat:@"<%08x %08x %08x %08x %08x %08x %08x %08x>",
                                  ntohl(tokenBytes[0]), ntohl(tokenBytes[1]), ntohl(tokenBytes[2]),
                                  ntohl(tokenBytes[3]), ntohl(tokenBytes[4]), ntohl(tokenBytes[5]),
                                  ntohl(tokenBytes[6]), ntohl(tokenBytes[7])];

        //self.pushDeviceToken = [credentials.token description];
        NSLog(@"Updating push device token for VOIP: %@",self.pushDeviceToken);
        //[TwilioVoice registerWithAccessToken:accessToken deviceToken:self.deviceTokenString completion:^(NSError *error) { }];
        [TwilioVoice registerWithAccessToken:self.accessToken deviceToken:self.pushDeviceToken completion:^(NSError * _Nullable error) {
            if (error) {
                NSLog(@"Error registering Voice Client for VOIP Push: %@", [error localizedDescription]);
            } else {
                NSLog(@"Registered Voice Client for VOIP Push");
            }
        }];
    }
}

I also had to change 2 lines to get the plugin to compile with the Twilio version.

Changed this:
        [TwilioVoice handleNotification:payload.dictionaryPayload delegate:self];
to:
        [TwilioVoice handleNotification:payload.dictionaryPayload delegate:self delegateQueue:NULL];

I'm not sure the delegateQueue should be NULL but so far my testing is working.

**Comment by vishalsaisystem** [comment]
Hi I am using this Plugin my Incoming Handler is not working while My outgoing is working fine , I am also getting notification from firebase, so anyone can help for solving the issue of that incoming handler is not getting fired
